tests/python_client/testcases/test_search.py

```diff
@@ -2617,7 +2618,76 @@ def test_search_with_consistency_session(self, nq, dim, auto_id, _async):
+    @pytest.mark.parametrize("index_name", ["_1ndeX$", "In_$0"])
+    def test_search_collection_naming_rules(self, nq, dim, collection_name, index_name, _async):
+        """
+        target: test search collection naming rules
+        method: 1. Connect milvus
+                2. Create a field with a name which uses all the supported elements in the naming rules
+                3. Create a collection with a name which uses all the supported elements in the naming rules
+                4. Create an index with a name which uses all the supported elements in the naming rules
+                5. Insert data (5000) into collection
+                6. Search collection
+        expected: searched successfully
+        """
+        nb = 5000
+        field_name1 = "_1nt"
+        field_name2 = "f10at_"
+        self._connect()
+        fields = [cf.gen_int64_field(), cf.gen_int64_field(field_name1),
+                  cf.gen_float_vec_field(field_name2, dim=dim)]
+        schema = cf.gen_collection_schema(fields=fields, primary_field=ct.default_int64_field_name)
+        collection_w = self.init_collection_wrap(name=collection_name, schema=schema,
+                                                 check_task=CheckTasks.check_collection_property,
+                                                 check_items={"name": collection_name, "schema": schema})
+        collection_w.create_index(field_name1, default_index_params, index_name=index_name)
+        int_values = pd.Series(data=[i for i in range(0, nb)])
+        float_vec_values = gen_vectors(nb, dim)
+        dataframe = pd.DataFrame({ct.default_int64_field_name: int_values,
+                                  field_name1: int_values, field_name2: float_vec_values})
+        collection_w.insert(dataframe)
+        collection_w.load()
+        vectors = [[random.random() for _ in range(dim)] for _ in range(nq)]
+        collection_w.search(vectors[:nq], field_name2, default_search_params,
+                            default_limit, _async=_async,
+                            check_task=CheckTasks.check_search_results,
+                            check_items={"nq": nq,
+                                         "limit": default_limit,
+                                         "_async": _async})
+
+    @pytest.mark.tags(CaseLabel.L1)
+    @pytest.mark.xfail(reason="issue 18951")
+    @pytest.mark.parametrize("partition_name", ["_Partii0n", "pArt1_i0n"])
+    def test_search_partition_naming_rules(self, nq, dim, auto_id, partition_name):
+        """
+        target: test search collection naming rules
+        method: 1. Connect milvus
+                2. Create a collection
+                3. Create a partition with a name which uses all the supported elements in the naming rules
+                4. Insert data into collection
+                5. with/without index with a name which uses all the supported elements in the naming rules
```

Comment:
This case only covers the search without index, needs another search case with index (index name uses naming rules)

Reply:
> This case only covers the search without index, needs another search case with index (index name uses naming rules)

The search case with index (index name uses naming rules) has been added.